website/content/docs/plugins/containerized-plugins.mdx

```diff
@@ -0,0 +1,253 @@
+registration time by using the `-command`, `-args`, and `-env` flags for
+`vault plugin register`.
+
+### Registration
+
+Once Vault has a suitable container runtime available, the remaining steps are
+very similar to registering and using other external plugins. The config setting
+for [`plugin_directory`](/vault/docs/configuration#plugin_directory) is not required
+for containerized plugins, but any plugin containers must already be available
+locally when mounted.
+
+1. Store the SHA256 of the plugin image:
+    ```bash
+    SHA256="$(docker images --no-trunc --format="{{ .ID }}" hashicorp/vault-plugin-secrets-kv | cut -d: -f2)"
+    ```
+
+1. Register a plugin, specifying your container image as `-oci_image`:
+    ```bash
+    vault plugin register \
+        -sha256="$SHA256" \
+        -oci_image=hashicorp/vault-plugin-secrets-kv \
+      secret kv-container
+    ```
+
+1. Enable the plugin:
+    ```bash
+    vault secrets enable kv-container
+    ```
+
+1. You can now interact with it like any other plugin:
+    ```shell-session
+    $ vault kv put -mount=kv-container hello subject=containers
+    Success! Data written to: kv-container/hello
+
+    $ vault kv get -mount=kv-container hello
+    ===== Data =====
+    Key        Value
+    ---        -----
+    subject    containers
+    ```
+
+### Container lifecycle and metadata
+
+Like any other external plugin, Vault will automatically manage the lifecycle
+of plugin containers. If they are killed out of band, Vault will restart it
+before servicing any requests that need to be handled by it. Vault will also
+[multiplex](/vault/docs/plugins/plugin-architecture#plugin-multiplexing) multiple
+mounts to be serviced by the same container if the plugin supports multiplexing.
+
+Vault labels each plugin container with a standard set of metadata to help
+identify the owner of the container, including the cluster ID, Vault's own
+process ID, and the plugin's name, type, and version.
+
+## Plugin runtimes
+
+Users who require more control over plugin containers can use the new "plugin
+runtime" APIs for finer grained settings. See the CLI documentation for
+[`vault plugin runtime`](/vault/docs/commands/plugin/runtime) for more details.
+
+## Supported configurations
+
+Vault's containerized plugins are compatible with a variety of configurations.
+In particular, it has been tested with the following:
+
+* Docker and Podman
+* Default and rootless installations
+* OCI runtimes runsc and runc
+* Plugin container images with root and non-root users
+
+Not all combinations work, and some have some limitations, listed below:
+
+### Rootless installation with non-root image user
+
+Not currently supported. We are hoping to provide limited support in future.
+
+### Rootless installation with runsc
+
+Works, but does not currently support cgroup limits, and requires  passing an
+additional `--ignore-cgroups` flag to the runsc runtime.
+
+## Podman
```

Comment:
If the process is as similar as we imply, we should fold the Podman steps into the guide rather than putting it at the end where it could easily be missed.

Reply:
My main hesitation here was that it felt overwhelming to present 4 different options concurrently. Would you prefer to split this into multiple smaller guides, or present it all together?

Reply:
The general rule of thumb is to keep things together and use tabs to separate versions of the same instructions (at least until we get variants for docs). It's better for SEO and it lets readers know their full set of options, instead of hoping they'll find a second version of the same doc with different instructions for the same thing.